Backend/database/repositories/filesystem.go

```diff
@@ -3,14 +3,87 @@ package repositories
 
 import (
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/docker/docker/client"
 )
 
+const volume_path = "/var/lib/documents/data"
+
 // Implements IRepositoryInterface
 type FilesystemRepository struct {
 	embeddedContext
 }
 
+type DockerFileystemRepository struct {
+	cli *client.Client
+}
+
+// Create instance of DockerFileSystemRepository struct
+func NewDockerFilesystemRespository() (c *DockerFileystemRepository, err error) {
+	c = new(DockerFileystemRepository)
+
+	c.cli, err = client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// Add file to volume or update if exists
+func (c *DockerFileystemRepository) AddToVolume(filename string) error {
+	// Check if source file is valid
+	src, err := os.Open(filename)
+	if err != nil {
+		return fmt.Errorf("Couldn't open source file")
+	}
+	// Create/update destination file and check it is valid
+	filepath := filepath.Join(volume_path, filename)
+	moved, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	defer moved.Close()
+	if err != nil {
+		return fmt.Errorf("Couldn't read/create the destination file")
+	}
+	// Copy source to destination
+	_, err = io.Copy(moved, src)
+	if err != nil {
+		return fmt.Errorf("File couldn't be copied to destination")
+	}
+	src.Close()
+	// Delete source file
+	err = os.Remove(filename)
+	if err != nil {
+		return fmt.Errorf("Couldn't remove the source file")
+	}
+	return nil
+}
+
+// Get file from volume. Returns a valid file pointer
+func (c *DockerFileystemRepository) GetFromVolume(filename string) (*os.File, error) {
+	// Concatenate volume path with file name
+	file, err := os.Open(filepath.Join(volume_path, filename))
+	if err != nil {
+		return nil, fmt.Errorf("File doesn't exist")
+	}
+	return file, nil
+}
+
+// Delete file from volume
+func (c *DockerFileystemRepository) DeleteFromVolume(filename string) error {
+	filepath := filepath.Join(volume_path, filename)
+	file, err := os.Open(filepath)
+	if err != nil {
+		return fmt.Errorf("File doesn't exist")
+	}
+	file.Close()
+	os.Remove(filepath)
```

Comment:
```suggestion
	if err = os.Remove(filename); err != nil {
		return fmt.Errorf("Couldn't remove the source file")
	}
```
`os.Remove` returns an error - make sure to include this error check to match code above